impeller/base/promise.h

```diff
@@ -17,6 +17,34 @@ std::future<T> RealizedFuture(T t) {
   return future;
 }
 
+// Wraps a std::promise and completes the promise with a value during
+// destruction if the promise does not already have a value.
+//
+// By default the std::promise destructor will complete an empty promise with an
+// exception. This will fail because Flutter is built without exception support.
+template <typename T>
+class PromiseDestructWrapper {
```

Comment:
I'm trying to think of a more descriptive name for this but coming up blank. `NoExceptionPromise`, `EmptyFulfillingPromise`? I guess this is fine.

Reply:
Renamed it to `NoExceptionPromise`